phonon/fields.py

```diff
@@ -0,0 +1,106 @@
+import itertools
+
+import phonon.connections
+
+
+class Field(object):
+
+    def key(self, *args):
+        return ".".join([str(a) for a in args])
+
+
+class ID(Field):
+
+    def __init__(self):
+        pass
+
+    def merge(self, a, b):
+        assert a == b
+        return a
+
+    def cache(self, *args):
+        return True
+
+
+class Sum(Field):
+
+    def __init__(self, dtype=int):
```

Comment:
dtype as a variable name seems a little opaque to me and I can't tell what it is, but I didn't look at the remaining code base :)